src/JWT/Algorithms/RSAlgorithmFactory.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Security.Cryptography.X509Certificates;
+
+namespace JWT.Algorithms
+{
+    public sealed class RSAlgorithmFactory : HMACSHAAlgorithmFactory
+    {
+        private readonly Func<X509Certificate2> _certFactory;
+
+        public RSAlgorithmFactory(Func<X509Certificate2> certFactory)
```

Comment:
Right. Since there will be an interface there now, it's up to consumers to implemented the way they need it. Since currently there is only 1 RS* algorithm and it requires `X509Certificate2` to function, I'll keep `RSAlgorithmFactory` title coupled with `RS256Algorithm`. Once there will be more, we can make the factory more generic. Makes sense?